docs/Coiote_IoT_DM/firmware_update.md

```diff
@@ -0,0 +1,188 @@
+
+1. Select **Basic Firmware Update**
+
+    ![Basic Firmware Update](images/Basic Firmware Update.png)
+
+1. Choose between **Pull** and **Push**
+
+    **Pull method** (recommended): The LwM2M Client receives the URI of the file that is to be downloaded and pulls the file from it. 
+    
+    **Push method**: The LwM2M Server pushes the firmware file to the device.
+
+
+    !!! Info
+        **Pull** supports any of the following **transport types**:
+
+        - `CoAP` or `CoAPs` over `UDP`
+        - `CoAP` or `CoAPs` over `TCP`
+        - `HTTP` or `HTTPs`
+
+        **Push** uses `CoAPs` over `UDP` by default.
+
+    !!! Note
+        **Which transport protocol to choose?**
+        
+        Downloads using `CoAP(s)` over `UDP` tend to be slow due to limitation of the maximum CoAP Block size of 1024 bytes and the required acknowledgements for each Block transfer.
+
+        Choosing `CoAP(s)` over `TCP` or `HTTP(s)` usually results in faster download speeds.
+
+
+    ![Basic Firmware Update](images/Pull.png)
+
+
+1. Click **Schedule Update** from the bottom-right corner to trigger the Firmware Update process.
+
+    ![Schedule Update](images/schedule.png)
+
+
+## Download & Upgrade Process
+
+If the Firmware Update is scheduled successfully, the device starts **downloading** the firmware at the next practical opportunity. The actual firmware **update** starts once the integrity and authenticity of the firmware image has been validated by the LwM2M Client.
+
+![Upgrading](images/update-in-progress.png)
+
+The status of the firmware update is shown in the Resources: **State** `/5/*/3` & **Update Result** `/5/*/5`. If no errors arise, the update process follows the following pattern:
+
+1. **Downloading** `state 1` & `update result 0` 
+2. **Downloaded** `state 2` & `update result 0` 
+3. **Updating** `state 3` & `update result 0` 
+4. **Updated** `state 0` & `update result 1` 
+
+
+!!! info
+    When the device is updating its firmware, it will deregister and reboot using the new firmware. This process may time multiple minutes.
+
+    ![Deregistered while upgrading](images/deregistered.png)
+
+
+If the update was successfully performed, the **State** `/5/*/3` returns to `0` and the **Update Results** `/5/*/5` reports `1`.
+
+![Firmware upgraded](images/result1.png)
+
+The **Update list** also reports: `Success`.
+
+![Successful update](images/success.png)
+
+
+
+## Troubleshooting
+
+* ### Insufficient flash
+
+    A common update error is due to insufficient flash memory, indicated by **Update Result** `2`.
+
+    The Anjay build file can be quite large in size, particularly when using the `.hex` file (`app_signed.hex`). For updating the firmware, it’s better to use the `app_update.bin` file which is about half the size of the `.hex` file.
+
+    !!! Warning
+        The **Nordic Thingy:91** is very restricted in its flash size due to the Nordic Bootloader. The maximum firmware size is `421kB`.
```

Comment:
This is actually the only place in the whole document where the Thingy:91 is referred to specifically. It feels oddly specific to single it out here, considering that the rest of the article is agnostic with regards to the hardware used.

Also please note that the `.hex` files are never going to work with Zephyr or Nordic devices. The `app_update.bin` is the only file that will work with Nordic devices. Phrases like "particularly when using the `.hex`" suggest otherwise.

If you want to single out the issues with Thingy:91, I believe this should be marked in a header-like fashion, much like the way you're talking about the Zephyr LwM2M Client below. Also, if you believe that using the `.hex` file instead of the `.bin` file may be a common mistake, then please describe it as such - once again, the `.hex` file will _never_ work (at least not on any of the platform known to us), even if it would otherwise fit, because it's not the right file format.

Reply:
Yeah, you're right. It wouldn't make sense to highlight the thingy:91 here, and maybe it's too obvious to mention people need a `.bin` file.

I'm adding a small note in one of the steps under: **Prepare the Firmware Update** 

```
1. **Upload the firmware image**, which is usually a binary (`.bin`) file. 

    !!! note 
        When using the Anjay LwM2M Client, the binary file is called `app_update.bin`.
```